account_journal_lock_date/models/account_move.py

```diff
@@ -0,0 +1,94 @@
+
+from odoo import api, fields, models, _
+
+from ..exceptions import JournalLockDateError
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    can_be_locked = fields.Boolean(
+        'Can be locked', compute="_compute_can_be_locked", store=True)
+    locked = fields.Boolean('Locked', readonly=True)
+
+    @api.multi
+    @api.depends('company_id', 'journal_id', 'date', 'locked')
+    def _compute_can_be_locked(self):
+        for move in self:
+            if move.locked:
+                move.can_be_locked = False
+            else:
+                fy_lock_date = move.company_id.fiscalyear_lock_date
+                journal_lock_date = move.journal_id.lock_date
+                lock = move.journal_id.permanent_lock
+                if lock:
+                    move.can_be_locked = bool(move.date >= journal_lock_date)
+                else:
+                    lock_date = max(fy_lock_date or '0000-00-00',
+                                    journal_lock_date or '0000-00-00')
+                    move.can_be_locked = bool(move.date >= lock_date)
+
+    @api.multi
+    def lock_move(self):
+        return self.sudo().write({'locked': True})
+
+    @api.multi
+    def unlock_move(self):
+        return self.sudo().write({'locked': False})
+
+    @api.multi
+    def _check_lock_date(self):
+        res = super(AccountMove, self)._check_lock_date()
+        if self.env['account.journal']._can_bypass_journal_lock_date():
+            for move in self:
+                lock_date = move.journal_id.lock_date
+                lock = move.journal_id.permanent_lock
+                if lock and lock_date and move.date <= lock_date:
+                    raise JournalLockDateError(
+                        _("The journal %s has been lock permanently "
+                          "at date %s. Please update lock date without "
+                          "permanency from the Lock Wizard.") %
+                        (move.journal_id.name, lock_date, ))
+            return res
+        for move in self:
+            lock_date = move.journal_id.lock_date
+            lock = move.journal_id.permanent_lock
+            if lock and lock_date and move.date <= lock_date:
+                raise JournalLockDateError(
+                    _("The journal has been lock permanently "
+                      "at date %s. Please update lock date without "
+                      "permanency from the Lock Wizard.") % (lock_date, ))
+            if lock_date and move.date <= lock_date:
+                raise JournalLockDateError(
+                    _("You cannot add/modify entries prior to and "
+                      "inclusive of the journal lock date %s") %
+                    (lock_date, ))
+            if move.locked:
+                raise JournalLockDateError(
+                    _("The move %s has been locked. You need to have "
+                      "'Adviser' role.") % move.name)
```

Comment:
For the test to work precisely, you need to have different exceptions. This last one is not a JournalLockDateError